syntax = "proto3";

/*
서버 상태를 전송하는 프로토콜임
Mac 주소를 키로 사용함
Float 는 소수점 2자리까지 사용
*/
message server_status {
  string mac = 1;
  string ip = 2;
  string hostname = 3;
  repeated float cpu_idle_rate = 4;  // CPU 는 여러개일 수 있으므로 repeated 로 처리
  float memory_rate = 5;
  map<string, uint64> projects = 6; // disk 는 partion, 용량으로 구성
}